project-management/src/main/java/life/qbic/projectmanagement/application/api/AsyncProjectService.java

```diff
@@ -1137,12 +1190,14 @@ public SampleRegistrationRequest(String projectId,
    * @param requests  a collection for {@link SampleRegistrationInformation} items
    * @since 1.10.0
    */
-  record SampleUpdateRequest(String projectId, Collection<SampleRegistrationInformation> requests) {
+  record SampleUpdateRequest(String projectId, Collection<SampleRegistrationInformation> requests,
+                             String requestId) {
 
-    public SampleUpdateRequest(String projectId,
-        Collection<SampleRegistrationInformation> requests) {
-      this.projectId = projectId;
-      this.requests = List.copyOf(requests);
+    public SampleUpdateRequest {
+      requireNonNull(projectId);
+      requestId =
+          isNull(requestId) || requestId.isBlank() ? UUID.randomUUID().toString() : requestId;
```

Comment:
i think this is a open door for nasty side effects. The client either provides the ID or must explicitly call the constructor method for creating an ID automatically.